docs/user_manual/serialization.md

```diff
@@ -0,0 +1,312 @@
+
+#### JSON schema homogeneous component data object
+
+A [`HomogeneousComponentData`](#json-schema-homogeneous-component-data-object) object contains the actual values of a certain component following the exact order of the attributes listed in the [`attributes`](#json-schema-root-object) field in the [`PowerGridModelRoot`](#json-schema-root-object) object.
+
+- [`HomogeneousComponentData`](#json-schema-homogeneous-component-data-object): `Array`
+  - [`AttributeValue`](#json-schema-attribute-value): the value of each attribute.
+
+#### JSON schema inhomogeneous component data object
+
+An [`InhomogeneousComponentData`](#json-schema-inhomogeneous-component-data-object) object contains actual values per attribute of a certain component.
+Contrary to the [`HomogeneousComponentData`](#json-schema-homogeneous-component-data-object), it lists the names of the attributes for which the values are specified, so the attributes may be in arbitrary order and do not have to follow the schema listed in the [`attributes`](#json-schema-root-object) field in the [`PowerGridModelRoot`](#json-schema-root-object) object.
+
+- [`InhomogeneousComponentData`](#json-schema-inhomogeneous-component-data-object): `Object`
+  - [`Attribute`](#json-schema-attribute): [`AttributeValue`](#json-schema-attribute-value): the value of each attribute per attribute.
+
+#### JSON schema attribute value
+
+The [`AttributeValue`](#json-schema-attribute-value) contains the actual value of an attribute. The value can be any of `null` if it is not provided, or any of [`int32_t`](#json-schema-int32_t), [`int8_t`](#json-schema-int8_t), [`double`](#json-schema-double), [`RealValueInput`](#json-schema-realvalueinput) or [`RealValueOutput`](#json-schema-realvalueoutput).
+The type is listed for each attribute in [Components](components.md).
+
+- [`AttributeValue`](#json-schema-attribute-value): `null` | [`int32_t`](#json-schema-int32_t) | [`int8_t`](#json-schema-int8_t) | [`double`](#json-schema-double) | [`RealValueInput`](#json-schema-realvalueinput) | [`RealValueOutput`](#json-schema-realvalueoutput)
+
+#### JSON schema int32_t
+
+An [`int32_t`](#json-schema-int32_t) is an integer `number` value usually representing an ID. It may be in the inclusive range $\left[-2^{31},+2^{31} - 1\right]$.
+The type is listed for each attribute in [Components](components.md).
+
+- [`int32_t`](#json-schema-int32_t): `number`
+
+#### JSON schema int8_t
+
+An [`int8_t`](#json-schema-int8_t) integer `number` value usually representing an enumeration value or a discrete setting. It may be in the inclusive range $\left[-2^{7},+2^{7} - 1\right]$.
+The type is listed for each attribute in [Components](components.md).
+
+- [`int8_t`](#json-schema-int8_t): `number`
+
+#### JSON schema double
+
+```{note}
```

Comment:
This rendering is not good here. Hyperlinks are not rendered. Maybe you do not want to put them in a code block. But just use emphasize font.